### ISSUE TITLE
More file formats support (ape, wav) + better mp2 handling

### ISSUE THREAD

**Issue by mp3splt** [mp3splt/mp3splt#82]
Imported from SourceForge on 2020-09-13 14:33:47
Created by **[rad-paul](https://sourceforge.net/u/rad-paul/)** on 2010-06-27 00:07:18
Original: https://sourceforge.net/p/mp3splt/feature-requests/71

---

Currently, your application are only one arround that offers a functionality to split music files through a GUI.

So, I found it logical to open a feature request here, despite the app name showing your intention to work only with mp3 files.

Feature request itself: Please implement support for more audio formats, like .flac, .ape and possibly even .wav.

There are a lot of scripts, some of them for Nautilus, but no option with a GUI.

I think that implementation of this feature would help to fill a vacuum in this area of GNU/Linux applications, helping whole GNU/Linux project.

Thank you, Paul.

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:33:48
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2010-07-01 21:19:10
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#68f4

---

Hello,

thank you for the feature request.

Mp3splt-project is not and will not be limited only to mp3 split, even if the name remained unchanged.
It currently also supports ogg vorbis files split.

Work is in progress to make the application more stable and flexible for future enhancements.
Libmp3splt (used by the command line program and the GUI) has now a "plugin" engine in order to support other formats.
Basically, to create a plugin for another format, a '.c' file needs to be created implementing some small functions (from which the most important is 'split') by using an external library to handle the real split on boundaries and tags.
The only problem is the lack of time.

For the ogg plugin, the 'splt_pl_*' functions are implemented:
 http://mp3splt.svn.sourceforge.net/viewvc/mp3splt/mp3splt-project/trunk/libmp3splt/plugins/ogg.c?revision=620&view=markup
The 'splt_pl_split' is the most important.

Regards,
-- 
Alex

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:33:50
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2012-10-10 21:09:53
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#7af7

---

- **summary**: Please add more file formats support --> More file formats support (flac, ape, wav)
- **milestone**:  --> Next_Release_(example)

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:33:53
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2013-04-28 21:10:53
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#44db

---

Experimental flac support is currently being added in the latest development version.
Anyone willing to test is welcome to use those snapshots:
 http://ioalex.net/testing_downloads/
Thank you.

-- 
Alex

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:33:56
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2013-07-21 21:59:46
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#49b5

---

Native FLAC support is now available from mp3splt version 2.6.

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:33:58
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2013-07-21 21:59:54
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#555b

---

- **summary**: More file formats support (flac, ape, wav) --> More file formats support (flac, ape,)

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:34:01
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2013-07-21 22:00:07
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#d990

---

- **summary**: More file formats support (flac, ape,) --> More file formats support (ape, wav)

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:34:04
Created by **[*anonymous](https://sourceforge.net/u/*anonymous/)** on 2014-01-03 10:51:35
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#484c

---

As I record my music from satellite radio that is coded in mp2, I would appreciate mp2 as another format, and I guess that I am not alone in this regard. (All DVB transport streams in Germany contain this format, as far as I know). Im Mp2DirectCut, the free library mpglib.dll is used. Maybe it can be used in mp3splt. It is available here, for instance: http://www.free-codecs.com/mp3DirectCut_download.htm. 

As mp2 and mp3 are basically the same, this should be comparatively easy, as compared to other formats. I found out that simply renaming an mp2 file into an mp3 file permits using Mp3spltPartable Version 2.2.0.0 to split such a file!

Thanks
Ekkehart

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:34:06
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2014-01-03 12:54:24
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#e644

---

Hello Ekkehart,

You have tried a recent version of mp3splt and it didn't cut the .mp2 files ?
If so, can you please give me a sample file at m@ioalex.net ?
I believe that it should also work with mp2 files.

Thank you.

-- 
Alex

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:34:09
Created by **[*anonymous](https://sourceforge.net/u/*anonymous/)** on 2014-01-03 14:35:55
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#e644/dfc1

---

Hi Alex,

thank you for your quick reaction!

I can split mp2 files, but the resulting file gets an mp3 extension. This is apparently a feature of mp3splt.exe. 

Mp3splt-gtk (Windows version) does not permit loading mp2 files. (I use Windows for my audio stuff because some of my devices do run under Linux.)

I could probably write a wrapper that does the renaming, and probably I shall do that, but I thought that the usefulness of the program for standard users could be improved with little effort.

Thank you for the useful work work!

Ekkehart

**Comment by mp3splt** [comment]
Imported from SourceForge on 2020-09-13 14:34:11
Created by **[io_alex_2004](https://sourceforge.net/u/io_alex_2004/)** on 2014-01-03 15:57:57
Original: https://sourceforge.net/p/mp3splt/feature-requests/71/#340a

---

I see. Thank you for reporting this issue.
